nautobot/core/settings_funcs.py

```diff
@@ -82,3 +84,84 @@ def parse_redis_connection(redis_database):
         return f"{redis_scheme}://{redis_creds}{redis_host}?db={redis_database}"
     else:
         return f"{redis_scheme}://{redis_creds}{redis_host}:{redis_port}/{redis_database}"
+
+
+def setup_structlog_logging(
+    django_logging: dict,
+    django_apps: list,
+    django_middleware: list,
+    log_level="INFO",
+    root_level="INFO",
+    debug=False,
+    debug_db=False,
+    plain_format=False,
+) -> None:
+    """Set up structlog logging for Django."""
+    django_logging["version"] = 1
+    django_logging["disable_existing_loggers"] = True
+    if "test" in sys.argv:
+        django_logging["handlers"] = {
+            "null_handler": {
+                "level": "INFO",
+                "class": "logging.NullHandler",
+            },
+        }
+        for logger in django_logging["loggers"].values():
+            logger["handlers"] = ["null_handler"]
+            logger["level"] = "INFO"
+
+    django_apps.append("django_structlog")
+    django_middleware.append("django_structlog.middlewares.RequestMiddleware")
```

Comment:
Is there a reason these can't/shouldn't be declared directly in `settings.INSTALLED_APPS` and `settings.MIDDLEWARE` instead of modifying them here?

Reply:
The reason is to separate structlog configuration to be able to use the same code in Cloud Portal, and possibly reuse it in managed services once it will be dependent on Nautobot version containing `setup_structlog_logging`.